src/main/java/teammates/ui/controller/InstructorStudentRecordsPageAction.java

```diff
@@ -78,8 +80,10 @@ public ActionResult execute() throws EntityDoesNotExistException {
                                                    StatusMessageColor.WARNING));
             }
         }
+        
+        boolean hasVisibleComments = checkForVisibleComments(giverEmailToCommentsMap);
 
-        if (sessions.isEmpty() && comments.isEmpty()) {
+        if (sessions.isEmpty() && !hasVisibleComments) {
```

Comment:
Do we still need `!hasVisibleComments`, could we just use `comments.isEmpty()` since `comments` has all the comments visible to the instructor (ie we won't be removing any element from `comments`)